### PR TITLE
Add Tests for Notification Page and Refactor Notifications-related Code

### PR DESCRIPTION
**Pull request by FlorianDinant**
### Unit and Integration Tests for Notification Page
- Added comprehensive unit tests for the Notification Page components.
- Covered core features, including notification rendering, interaction handling (e.g., marking as read/unread, delete notifications using swipe etc...).
- Included tests for edge cases such as an empty notification list, notifications with different types, and handling.
### Refactoring of notifications.tsx
- Cleaned up the component code by removing unused imports and optimizing function implementations.
- Remove unnecessary comments
- Improved prop types
### Enhancements to Notifications API Integration:
- Updated API call logic to handle loading states, error handling, and retry mechanisms.
- Improved response data processing for smoother user experience.
### Other Improvements:
- Ensured all notification components adhere to consistent styling and design guidelines.

## Tests Added
- Notification Rendering: Verifies that notifications are displayed correctly based on the API response.
- Mark as Read Functionality: Tests that the "mark as read" action updates the UI and triggers the correct API call.
- Pagination/Infinite Scrolling: Checks that additional notifications are loaded correctly when the user scrolls to the end of the list.
- Error Handling: Ensures that appropriate error messages are shown when there is a failure to fetch notifications.

### Edweiss-app/
- edweiss-app/__test__/components/notifs/**NotifDisplay.test.tsx** : component displaying a notification
- edweiss-app/__test__/notifs/**NotificationsPage.test.tsx** : Main component file displaying the full page

## Checklist/
✅ Added unit and integration tests for the Notification Page.
✅ Refactored notifications.tsx for better readability and performance.
✅ Verified that all existing and new tests pass.
✅ Updated relevant documentation and code comments.

Please review the changes and provide any feedback.

### CODE REVIEW

File: edweiss-app/app/(app)/(tabs)/notifications.tsx [hunk 1]
```diff
@@ -96,85 +111,74 @@ const NotificationsTab: ApplicationRoute = () => {
...
                         {notifsDay.length > 0 && (
-                            <>
-                                <TText mb={10} mt={15} ml={2} size={16} color='darkBlue' bold={true}>{t(`notifications:today`)}</TText>
+                            <TView testID={testIDs.todayView}>
+                                <TText testID={testIDs.todayText} mb={10} mt={15} ml={2} size={16} color='darkBlue' bold={true}>{t(`notifications:today`)}</TText>
                                 {notifsDay.map((notif, index) => (
-                                    <TView key={index} testID={testIDs.notifView}>
-                                        <NotifDisplay item={notif.data} id={notif.id} dateSection='today' index={notifs.indexOf(notif)} />
-                                    </TView>
+                                    <NotifDisplay item={notif.data} id={notif.id} dateSection='today' index={index} key={notif.data.title} />
```
Comment (TuanDangNguyen2003) [inline edweiss-app/app/(app)/(tabs)/notifications.tsx:132]: There is just a warning about the `key` prop. When you .map on each item and then `Display` it then React-native will take into account the `key` of each displayed item. So if the `key`s between 2 different items are identical, react-native will show a warning. Even though it does not affect the usability or display-ability of the code. Just to be safe, we must keep in mind to have a clean, without-warning product at the end.

**Review by TuanDangNguyen2003 — APPROVED** [review]
# Code Review

### Compliments

- **Refactoring and Cleanup:** Great work on the consistent efforts to refactor and clean up the code, particularly in `NotifDisplay.tsx`. The removal of unnecessary comments and the introduction of constants contribute significantly to readability and maintainability.

- **Modularization:** The organization of constants for icons and colors within `NotifDisplay.tsx` is excellent. Also you have moved action functions to a separate file which makes `NotifDisplay.tsx` looks very clear. This modular approach makes the code more manageable and easier to extend in the future.

- **Testing:** The addition of comprehensive tests for notification components and utility functions reflects a strong commitment to code reliability and functionality. This level of testing is valuable for identifying potential issues early and maintaining high code quality.

### Areas for Improvement

- **Redundancy in Test Data:** Some redundancy appears in the mock data for notifications across test files. Consider using shared setup functions or fixtures to reduce duplication and streamline your test data setup. But I do understand that mockings is annoying so if you have the time, you can group them to avoid duplicated mocks. It is definitely just a minor problem.

- **Error Handling:** While the tests cover successful cases well, adding tests for error scenarios could improve robustness. This way, potential issues are gracefully managed, ensuring a smoother user experience.

- **Code Comments:** Including more descriptive comments, especially around complex logic or critical sections, would be beneficial. These annotations will help future developers or reviewers (me xD) better understand the purpose and flow of the code.

### Potential Issues

- **Key Prop Warning:** Double-check that all key props in lists and maps are unique and stable. Ensuring this prevents potential React warnings or bugs related to list rendering.

### Summary

The code improvements demonstrate a clear focus on enhancing quality and maintainability. Keep up the good work, and consider the suggested enhancements to make the codebase even more robust!
